sympy/solvers/inequalities.py

```diff
@@ -417,7 +417,7 @@ def solve_univariate_inequality(expr, gen, relational=True):
             solns = solve(n, gen, check=False)
             singularities = solve(d, gen, check=False)
         else:
-            solns = solve(e, gen, check=False)
+            solns = solve(e, gen, check=True)
```

Comment:
@smichr , due to check=False wrong solution to the equation `(2 + sqrt(x - 4)` being returned and the function _real_conditions wasnt being called. Now, correct solution is being returned i.e. `(4, oo)`